Libraries/LibWeb/HTML/Scripting/Agent.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2025, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/Vector.h>
+#include <LibGC/Root.h>
+#include <LibJS/Forward.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/CustomElements/CustomElementReactionsStack.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/webappapis.html#similar-origin-window-agent
+struct Agent {
+    GC::Root<HTML::EventLoop> event_loop;
+
+    // FIXME: These should only be on similar-origin window agents, but we don't currently differentiate agent types.
+
+    // https://dom.spec.whatwg.org/#mutation-observer-compound-microtask-queued-flag
+    bool mutation_observer_microtask_queued { false };
+
+    // https://dom.spec.whatwg.org/#mutation-observer-list
+    // FIXME: This should be a set.
+    Vector<GC::Ref<DOM::MutationObserver>> mutation_observers;
```

Comment:
Hmm, I think CI might be complaining from this not being visited? I'm not sure what makes this any different from a GC perspective though..? It was a `Vector<GC:Ref<T>>` beforehand, so not sure how this would become a problem now

Reply:
yeah that turned out to be the issue, wonder why it didnt come up before